test/com/ableton/PythonBuilderTest.groovy

```diff
@@ -0,0 +1,103 @@
+package com.ableton
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertNotNull
+
+import com.lesfurets.jenkins.unit.BasePipelineTest
+import org.junit.Before
+import org.junit.Test
+
+
+class PythonBuilderTest extends BasePipelineTest {
+  @SuppressWarnings('FieldTypeRequired')
+  def script
+
+  @Override
+  @Before
+  void setUp() throws Exception {
+    super.setUp()
+
+    this.script = loadScript('test/resources/EmptyPipeline.groovy')
+    assertNotNull(script)
+
+    helper.with {
+      registerAllowedMethod('error', [String], JenkinsMocks.error)
+      registerAllowedMethod('deleteDir', [], null)
+      registerAllowedMethod('httpRequest', [Map], null)
+      registerAllowedMethod('isUnix', [], JenkinsMocks.isUnix)
+      registerAllowedMethod('pwd', [Map], JenkinsMocks.pwd)
+      registerAllowedMethod('sh', [String], JenkinsMocks.sh)
+    }
+  }
+
+  @Test
+  void install() throws Exception {
+    PythonBuilder pb = new PythonBuilder(script: script, version: '1.0.0')
+    String pythonPath = pb.tempDir() + '/' + pb.pythonBaseName()
+    JenkinsMocks.addShMock('tar xfz ' + pythonPath + '.tgz', '', 0)
+    JenkinsMocks.addShMock('./configure --prefix=' + pythonPath, '', 0)
+    JenkinsMocks.addShMock('make -j1', '', 0)
+    JenkinsMocks.addShMock('make install', '', 0)
+    assertEquals(pythonPath, pb.install())
+  }
+
+  @Test(expected = Exception)
+  void installWithBuildError() throws Exception {
+    PythonBuilder pb = new PythonBuilder(script: script, version: '1.0.0')
+    String pythonPath = pb.tempDir() + '/' + pb.pythonBaseName()
+    JenkinsMocks.addShMock('tar xfz ' + pythonPath + '.tgz', '', 0)
+    JenkinsMocks.addShMock('./configure --prefix=' + pythonPath, '', 0)
+    JenkinsMocks.addShMock('make -j1', '', 1)
+    JenkinsMocks.addShMock('make install', '', 0)
```

Comment:
Is it really necessary to add that sh mock for `make install`?

Reply:
No, it's not, so I will remove this. FYI this was necessary with jenkins-pipeline-mocks 0.2.0 (which I started out with when I wrote these tests), but in 0.4.1 we fixed `sh`'s exception throwing behavior so it is no longer necessary.